modules/administration-guide/pages/upgrading-che-7-41-on-openshift.adoc

```diff
@@ -11,67 +11,85 @@ The workspace engine and authentication system used up to and including {prod-pr
 
 .Prerequisites
 
-* An instance of {prod-prev-short} deployed on xref:supported-platforms.adoc[], using the default internal PostgreSQL database, and with OpenShift OAuth enabled. See {prod-docs-url-enable-oauth}.
-* Command line tools `oc`, `curl`, and `jq` are available.
+* An instance of {prod-prev-short} deployed on one of the xref:supported-platforms.adoc[], using the default internal PostgreSQL database, and with OpenShift OAuth enabled. See {prod-docs-url-enable-oauth}.
+* The following CLI tools are available:
+** `oc`
+** `curl`
+** `jq`
 * The host running the upgrade commands is running on Linux.
+* Optional:
+.. All changes from all workspaces have been commited and pushed to their Git remotes.
+.. All workspaces have been stopped to avoid UX degradation.
+.. The {prod-prev-short} data have been backed up. See {prod-docs-url-backup-recovery}.
```

Comment:
- Would these be better as first steps in the procedure? 
- Instead of saying "optional," you can preface the step with a condition (precondition). That way, the user can decide if the condition applies to their situation.  
- Some steps, like backing up data, are so common they might not need the precondition.

For example:

```suggestion
* Optional:
. To avoid losing work, commit and push changes from all workspaces to their Git remotes.
. To avoid impacting users' experiences, stop all workspaces.
. Back up the {prod-prev-short} data. See {prod-docs-url-backup-recovery}.
```

Reply:
Rolfe, i like the wording in your suggestion. However, the third/last step would only apply to those admins who are already using the referenced functionality, which means it's definitively optional.

These steps were originally, as you can see in the diff, in the procedure as one first step, listed as a series in one sentence.
However, i'd like to avoid lumping all non-migration tasks and migration tasks into one procedure.
So I figured it'd make things clearer if i moved those into the prerequisites.

Reply:
Also, i'm not worried about the fact that with this layout and these extra points the list of prerequisites is quite long. This is a very important procedure, so it's something that has to be approached with extra care anyway.

Reply:
@max-cx Sure, keeping it in the prerequisites is fine. We still need to inform the user of the basis for their choice. Stating that it's optional isn't enough.

Reply:
It sounds like you should provide the user with a precondition for the third/last prerequisite.